targets/pna_nic/thrift/src/PnaNic_server.cpp

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2024 Marvell Technology, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Rupesh Chiluka (rchiluka@marvell.com)
+ *
+ */
+
+#include <bm/config.h>
+#include <bm/PnaNic.h>
+
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/server/TSimpleServer.h>
+#include <thrift/transport/TServerSocket.h>
+#include <thrift/transport/TBufferTransports.h>
+
+namespace thrift_provider = apache::thrift;
+
+#include <bm/bm_sim/switch.h>
+#include <bm/bm_sim/logger.h>
+#include <bm/thrift/stdcxx.h>
+
+#include "pna_nic.h"
+
+using namespace bm::pna;
+
+namespace pnic_runtime {
+
+class PnaNicHandler : virtual public PnaNicIf {
+ public:
+  explicit PnaNicHandler(PnaNic *sw)
+    : nic_(sw) { }
+  // }
+
+  int64_t get_time_elapsed_us() {
+    bm::Logger::get()->trace("get_time_elapsed_us");
+    // cast from unsigned to signed
+    return static_cast<int64_t>(nic_->get_time_elapsed_us());
+  }
+
+  int64_t get_time_since_epoch_us() {
+    bm::Logger::get()->trace("get_time_since_epoch_us");
+    // cast from unsigned to signed
+    return static_cast<int64_t>(nic_->get_time_since_epoch_us());
+  }
+
+ private:
+  PnaNic *nic_;
+};
+
+stdcxx::shared_ptr<PnaNicIf> get_handler(PnaNic *sw) {
```

Comment:
Nit: I guess that in `simple_switch` code `sw` in `SimpleSwitch *sw` means "switch"? Might consider using a different variable name here.

Reply:
Changed to `nic`.